src/sql/workbench/browser/editor/tableDesigner/tableDesignerInput.ts

```diff
@@ -42,6 +56,7 @@ export class TableDesignerInput extends EditorInput {
 				this._onDidChangeDirty.fire();
 			}
 		}));
+		this._tableIcon = tableInfo.tableIcon.toString() === '' ? TableIcon.Basic : tableInfo.tableIcon as TableIcon;
```

Comment:
this will throw if the tableInfo.tableIcon is undefined. why not just use `!tableInfo.tableIcon?`

Reply:
Or even easier `tableInfo.tableIcon ?? TableIcon.Basic`